Brokerages/DefaultOrderBook.cs

```diff
@@ -153,21 +161,36 @@ public void RemoveBidRow(decimal price)
         /// <param name="price">The ask price level to be removed</param>
         public void RemoveAskRow(decimal price)
         {
-            lock(_locker)
+            lock (_locker)
             {
-                _asks.Remove(price);
+                Asks.Remove(price);
             }
 
             if (price == BestAskPrice)
             {
-                lock(_locker)
+                lock (_locker)
                 {
-                    BestAskPrice = _asks.Keys.FirstOrDefault();
-                    BestAskSize = BestAskPrice > 0 ? _asks[BestAskPrice] : 0;
+                    BestAskPrice = Asks.Keys.FirstOrDefault();
```

Comment:
I think you should be able to do `var kvp = Asks.FirstOrDefault()` and then use that to assign `BestAskPrice` and `BestAskSize` -- this precludes use from needing to access the dictionary twice. Same comment for the bids.